app/packages/core/src/plugins/SchemaIO/utils/generate-schema.ts

```diff
@@ -16,10 +16,9 @@ export function generateSchema(value, options?) {
           label,
           component: "TableView",
           columns: getTableColumns(value),
-          readOnly: true,
         },
       };
-    } else if (isPrimitiveType(dominantType) && readOnly) {
+    } else if (isPrimitiveType(dominantType || "") && readOnly) {
```

Comment:
This is probably the crux of the fix, but I find it very hard to follow. Its odd that `getDominantType()` returns null(ish) values? A variable name or comment indicating that the `||` is due to it being an empty array (isn't that the only case where this happens?

Also I realize we're using existing utils here, but I'd rather have metadata be in object form than functions, the data is already there so its a bit odd to call functions to get at it:

```typescript
const value = "foo";
const typeInfo = getTypeInfoForValue(value);
// typeInfo.isArray => false
// typeInfo.dominantType => null (rename to typeInfo.arrayItemDominantType)
// typeInfo.isPrimitive => true
// typeInfo.value => "foo"
// typeInfo.isNullish => false
// typeInfo.type => string
// etc
```

Both of these a minor comments, but thought they might be useful feedback.

Reply:
Yah, that's right. dominantType will be unknown for an empty array. Planning to use something like one of view in future for unknown type but it can potentially be a nested object.

Not too sure about the second half of the comment. The `generate-schema` (I should probably rename to `inferSchemaUsingValue(value)` is only used for `InferredView` to generate a `SchemaIO` compatible schema based only on a value.